nncf/torch/quantization/quantize.py

```diff
@@ -0,0 +1,150 @@
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import deepcopy
+from typing import Optional, Any, Tuple, Dict
+
+import torch
+
+from nncf import Dataset
+from nncf import NNCFConfig
+from nncf import QuantizationPreset
+from nncf import TargetDevice
+from nncf.config.structures import BNAdaptationInitArgs
+from nncf.config.structures import QuantizationRangeInitArgs
+from nncf.torch.dynamic_graph.io_handling import wrap_nncf_model_inputs_with_objwalk
+from nncf.torch.dynamic_graph.io_handling import wrap_nncf_model_outputs_with_objwalk
+from nncf.torch.initialization import PTInitializingDataLoader
+from nncf.torch.model_creation import create_compressed_model
+from nncf.torch.nested_objects_traversal import objwalk
+from nncf.torch.utils import get_model_device
+from nncf.torch.utils import is_tensor
+
+
+# TODO(alexsu52): It is a workaround and should be removed.
+class CalibrarionDataLoader(PTInitializingDataLoader):
+    """
+    This class wraps the nncf.Dataset.
+
+    This is required for proper initialization of certain compression algorithms.
+    """
+
+    def __init__(self, data_loader: Dataset):
+        super().__init__(data_loader)
+        self._length = None
+
+    @property
+    def batch_size(self):
+        return getattr(self._data_loader.data_source, 'batch_size', 1)
+
+    def __iter__(self):
+        return iter(self._data_loader.get_inference_data())
+
+    def __len__(self):
+        if self._length is None:
+            data = self._data_loader.get_inference_data()
+            self._length = CalibrarionDataLoader._get_length(data)
+        return self._length
+
+    def get_inputs(self, dataloader_output: Any) -> Tuple[Tuple, Dict]:
+        if not isinstance(dataloader_output, tuple):
+            dataloader_output = (dataloader_output, )
+        return dataloader_output, {}
+
+    @staticmethod
+    def _get_length(iterable) -> int:
+        length = 0
+        for _ in iterable:
+            length = length + 1
+
+        return length
+
+
+def quantize_impl(model: torch.nn.Module,
+                  calibration_dataset: Dataset,
+                  preset: QuantizationPreset,
+                  target_device: TargetDevice,
+                  subset_size: int,
+                  fast_bias_correction: bool,
+                  model_type: Optional[str] = None) -> torch.nn.Module:
+    """
+    Implementation of the `quantize()` method for the PyTorch backend.
+    """
+    if model_type is not None:
```

Comment:
Can we support `model_type` here as well and add the following lines in the config file that are introduced here: https://github.com/huggingface/optimum-intel/blob/c0cefed96f8d3a0e9b39289a0f11b9ef60088f46/optimum/intel/openvino/configuration.py#L30.
This was validated on quite a lot of models already and works pretty well.

Reply:
Please take a look.